app/js/invertedIndex.js

```diff
@@ -1,67 +1,106 @@
+/**
+ * InvertedIndex class
+ * @class
+ */
 class InvertedIndex {
-
-  constructor(){
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
     this.indexed = {};
   }
 
+  /**
+   * Get the index for a particular file
+   * @function
+   * @param {String} fileName
+   * @return {Object} index object
+   */
   getIndex(fileName) {
-      return this.indexed[fileName];
+    return this.indexed[fileName];
   }
 
-  readFile(file) {
+  /**
+   * Read files using FileReader
+   * @function
+   * @param {String} file
+   * @return {Promise} validateFile response
+   */
+  static readFile(file) {
     return new Promise((resolve, reject) => {
       try {
         const reader = new FileReader();
-        reader.onload = (e) => {
+        reader.onload = () => {
           const fileToValidate = JSON.parse(reader.result);
-            let response = this.validateFile(fileToValidate);
-            resolve(response);
-        }
-        reader.readAsText(file)
-      }
-      catch(error) {
+          const response = InvertedIndex.validateFile(fileToValidate);
+          resolve(response);
+        };
+        reader.readAsText(file);
+      } catch (error) {
         reject(error);
       }
     });
   }
 
-  validateFile(fileToValidate) {
+  /**
+   * Validate File
+   * @function
+   * @param {string} fileToValidate content of the file uploaded
+   * @return {Object} result.success should return true or false
+   */
+  static validateFile(fileToValidate) {
     const fileLength = fileToValidate.length;
-    for(let key = 0; key < fileLength; key += 1) {
+    let result = {};
+    for (let key = 0; key < fileLength; key += 1) {
       if (typeof fileToValidate !== 'object'
           || Object.keys(fileToValidate[key]).length !== 2
           || fileToValidate[key].title === undefined
           || fileToValidate[key].text === undefined
           || typeof fileToValidate[key].title !== 'string'
           || typeof fileToValidate[key].text !== 'string') {
-            return {
-              success: false,
-              message: 'has an invalid JSON format.'
-            };
+        result = {
+          success: false,
+          message: 'has an invalid JSON format.'
+        };
       } else {
-          return {
-            success: true,
-            message: 'File is valid',
-            fileToValidate
-          };
+        result = {
+          success: true,
+          message: 'File is valid',
+          fileToValidate
+        };
       }
     }
+    return result;
   }
 
-  tokenize(words) {
+  /**
+   * Get individual words from a string of text.
+   * @function
+   * @param {String} words text to be tokenized.
+   * @return {Array} array of string tokens
+   */
+  static tokenize(words) {
     const pattern = /[ .:;?!~,`'&|()<>{}[\]\r\n/\\]+/;
     return words.toLowerCase().split(pattern);
   }
```

Comment:
Thanks a lot. I actually think numbers should not be removed. An example is 7th, if this is included, it will take away 7 and you will be left with 'th' which will not make sense to the user.
But noted too.